### DISCUSSION TITLE
Start recording on first MIDI-note input

### DISCUSSION THREAD

**Discussion by KnightWhoSaysNeeeowWumPing** [hydrogen-music/hydrogen#1066] [Q&A]
Hello there!

I use hydrogen for MIDI-recording of an electronic drum set and was wondering if there is any way to start recording only when I hit a drum pad. Currently I'm having a hard time to time my recordings right.
My workflow is the following: I play along some songs (directly over my e-drum controller) and record its MIDI-output in hydrogen. It's just a real pain to time the start of the recording correctly so that every measure is within the correct frame.

Another workaround would be to attach the song file (e.g. mp3) to my hydrogen project so that it aligns automatically and I didn't have to worry about it being on point - but I'm afraid this is not possible, is it?

Is it furthermore possible to shift a finished recording session (namely its notes) altogether (left or right) to fix such a badly timed recording?

Thanks for your help and for this great software in general!

**Comment by mauser** [comment]
Hi! 

to be honest, i never used that midi recording feature of Hydrogen, so i can't comment on its details. I know that you can enable a second, hidden menu if you right click on that record button, but i have no idea what that is about :(

But it would be possible to "attach" the song to Hydrogen. Hydrogen has a "playback track" feature since Hydrogen 1.0 (briefly described [here](http://hydrogen-music.org/documentation/manual/manual_en.html#sect.song_editor.playbackTrack). You can use all file formats which Hydrogen supports for its samples (wav, flac, ogg..). At the moment you can't "move" that track, so you may need to add a count-in part or sth. at the beginning of your audio file for orientation. 

The shifting of recording notes would be only possible on a per pattern basis, the current git version of Hydrogen (what will become 1.1) supports that. This version support multi-selecting notes by dragging a "lasso" around them.

**Reply by KnightWhoSaysNeeeowWumPing** [reply]
Hey mauser,

thank you for the quick reply!
This sounds very promising - I'm going to try the "playback track" feature (a shame I didn't find it myself) and look into the hidden record menu.
As a side note I forgot to check the version of hydrogen I'm running on my recording machine... a shame that on the **ubuntu studio** that I'm using, the repositories hold only the 0.9.7 version from 2016 - so I'll install the current one manually. Maybe it's time to change my recording machine to Arch, too :)

Thank you again, I'll keep you posted

**Reply by theGreatWhiteShark** [reply]
> As a side note I forgot to check the version of hydrogen I'm running on my recording machine... a shame that on the ubuntu studio that I'm using, the repositories hold only the 0.9.7 version from 2016 - so I'll install the current one manually. Maybe it's time to change my recording machine to Arch, too :)

AFAIK ubuntu studio 20.04 bundles `1.0.0~rc1` release. But current master should be better anyway :wink:
